src/main/java/org/testng/internal/Invoker.java

```diff
@@ -444,6 +444,10 @@ private void invokeConfigurationMethod(Object targetInstance,
       ConstructorOrMethod method = tm.getConstructorOrMethod();
 
       IConfigurable configurableInstance = computeConfigurableInstance(method, targetInstance);
+      if (RuntimeBehavior.isDryRun()) {
```

Comment:
I understand your change has very few modification but I'm not fan of global var. What will be the effect of passing a `RuntimeBehavior` as constructor param?